sdk/python/setup.py

```diff
@@ -92,13 +92,14 @@ def read_readme() -> str:
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
         'Topic :: Scientific/Engineering',
         'Topic :: Scientific/Engineering :: Artificial Intelligence',
         'Topic :: Software Development',
         'Topic :: Software Development :: Libraries',
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
-    python_requires='>=3.7.0,<3.12.0',
+    python_requires='>=3.7.0,<3.13.0',
```

Comment:
Since the upper bound Python version is decoupled from changing the recommended minimum Python version, let's make both changes in the same PR.

Additionally, this upper bound is a bit risky since we don't actually know that the _current_ KFP SDK on `master:HEAD` (and subsequent releases) will be compatible with Python 3.12, since it's not generally available yet nor tested on a pre-release version. Once Python 3.12 is released, users of KFP SDK 2.3.0 (the next release), for example, might try to upgrade to Python 3.12 and find it does not work.